src/pipeline.cc

```diff
@@ -825,4 +826,61 @@ Result Pipeline::GenerateOpenCLLiteralSamplers() {
   return {};
 }
 
+Result Pipeline::GenerateOpenCLPushConstants() {
+  if (!IsCompute() || GetShaders().empty() ||
+      GetShaders()[0].GetShader()->GetFormat() != kShaderFormatOpenCLC) {
+    return {};
+  }
+
+  const auto& shader_info = GetShaders()[0];
+  if (shader_info.GetPushConstants().empty())
+    return {};
+
+  // Determine size and contents of the push constant buffer.
+  std::vector<uint8_t> bytes;
+  for (const auto& pc : shader_info.GetPushConstants()) {
+    if (pc.offset + pc.size > bytes.size()) {
+      bytes.resize(pc.offset + pc.size);
+    }
+
+    switch (pc.type) {
+      case Pipeline::ShaderInfo::PushConstant::kPushConstantDimensions:
+        // All compute kernel launches are 3D.
+        assert(pc.size == 4);
+        *reinterpret_cast<uint32_t*>(bytes.data() + pc.offset) = 3;
```

Comment:
Would it be simpler to just make the buffer a `uint32_t` and then multiply the size  on line 872 by `sizeof(uint32_t)` ?

Reply:
Sort of. I'm trying to make it simple for us to support values that are not 32-bits in the future (since we may use push constants for kernel arguments at some point). I've made something that resembles your suggested change though, as it does simplify some logic.